### DISCUSSION TITLE
How can I download my certificates?

### DISCUSSION THREAD

**Discussion by joefoxton** [vercel/vercel#5192] [Help]
I need to upload a certificate to a HAProxy, so I can expose our Minio S3 storage publicly via https, allowing the Veeam backup software to use my storage as a backup destination. Veeam will only connect with storage published via https, and signed by a valid certificate. Vercel currently runs my domain...

So how so I download an existing certificate & private key so it can be uploaded to my HAProxy?

If that's not possible, how can I ssh into Vercel to create a new one?

**Comment by paulogdm** [comment]
Certificates issued by Vercel can only be used on our platform. If you are using the domain elsewhere, you need to check ways of adding a certificate in that external server. The server is responsible for serving a certificate, not the DNS provider nor the registrar.

If you need to upload a custom certificate to Vercel, an enterprise plan is required. However, I do need to say that for the majority of use-cases, that is not required at all.
